src/main/tray.ts

```diff
@@ -0,0 +1,86 @@
+import { app as rawApp, ipcMain, Menu, Tray } from 'electron'
+import { globalShortcut } from 'electron'
+import path from 'path'
+import * as mainWindow from './windows/main'
+import { ExtendedAppMainProcess } from './types'
+import { getLogger } from '../shared/logger'
+import { appIcon } from './application-constants'
+
+let tray: Tray = null
+let contextMenu: Menu = null
+
+const app = rawApp as ExtendedAppMainProcess
+const log = getLogger('main/tray')
+
+ipcMain.on('updateTrayIcon', updateTrayIcon)
+
+export function updateTrayIcon() {
+  const tx = app.translate
+
+  const IsMac = process.platform === 'darwin'
+  if (!IsMac) {
```

Comment:
Can't we do an early return here?

`if (process.platform === 'darwin') return`

Reply:
Would love to have less indentation levels here if possible